transit_odp/organisation/tests/csv/test_service_codes.py

```diff
@@ -121,92 +138,308 @@ def test_registered_service_code():
         licence=otc_lic,
         registration_number=service_codes[1],
         service_number=service_numbers[1],
+        effective_date=datetime.datetime(2023, 6, 24),
+    )
+    SeasonalServiceFactory(
+        licence=bods_licence,
+        registration_code=service_codes[1][-1:],
+        start=datetime.datetime(2022, 2, 24),
+        end=datetime.datetime(2024, 2, 24),
     )
+
+    dataset3 = DatasetFactory(organisation=org1)
+    # Require Attention: Yes
+    # operating_period_end_date is not None
+    TXCFileAttributesFactory(
+        revision=dataset3.live_revision,
+        service_code=service_codes[3],
+        operating_period_end_date=datetime.datetime(2023, 2, 24),
+        modification_datetime=datetime.datetime(2022, 6, 24),
+    )
+    # staleness_otc = True => "Stale - OTC Variation"
     ServiceModelFactory(
         licence=otc_lic,
-        registration_number=service_codes[0],
-        service_number=service_numbers[0],
+        registration_number=service_codes[3],
+        service_number=service_numbers[3],
+        effective_date=datetime.datetime(2023, 3, 24),
+    )
+    SeasonalServiceFactory(
+        licence=bods_licence,
+        registration_code=service_codes[3][-1:],
+        start=datetime.datetime(2022, 2, 24),
+        end=datetime.datetime(2024, 2, 24),
+    )
+
+    dataset4 = DatasetFactory(organisation=org1)
+    # Require Attention: Yes
+    # operating_period_end_date is not None
+    TXCFileAttributesFactory(
+        revision=dataset4.live_revision,
+        service_code=service_codes[4],
+        operating_period_end_date=datetime.datetime(2023, 3, 24),
+        modification_datetime=datetime.datetime(2023, 6, 24),
     )
+    # staleness_end_date = True => "Stale - End date passed"
     ServiceModelFactory(
         licence=otc_lic,
-        registration_number=service_codes[2],
-        service_number=service_numbers[2],
+        registration_number=service_codes[4],
+        service_number=service_numbers[4],
+        effective_date=datetime.datetime(2022, 6, 24),
+    )
+    SeasonalServiceFactory(
+        licence=bods_licence,
+        registration_code=service_codes[4][-1:],
     )
 
-    mock_writer = Mock()
-    mock_csv_writer.return_value = mock_writer
+    dataset5 = DatasetFactory(organisation=org1)
+    # operating_period_end_date is not None
+    TXCFileAttributesFactory(
+        revision=dataset5.live_revision,
+        service_code=service_codes[5],
+        operating_period_end_date=datetime.datetime(2023, 6, 24),
+        modification_datetime=datetime.datetime(2022, 1, 24),
+    )
+    # staleness_12_months_old = True => "Stale - 12 months old"
+    ServiceModelFactory(
+        licence=otc_lic,
+        registration_number=service_codes[5],
+        service_number=service_numbers[5],
+        effective_date=datetime.datetime(2021, 1, 24),
+    )
+    SeasonalServiceFactory(
+        licence=bods_licence,
+        registration_code=service_codes[5][-1:],
+    )
```

Comment:
Bit risky not specifying a start and end date - the factory might randomly generate dates that make today in season!

Applies in a few cases below too.